session/session.go

```diff
@@ -1384,6 +1385,13 @@ func (s *session) ParseWithParams(ctx context.Context, sql string, args ...inter
 	for _, warn := range warns {
 		s.sessionVars.StmtCtx.AppendWarning(util.SyntaxWarn(warn))
 	}
+	if variable.TopSQLEnabled() {
+		normalized, digest := parser.NormalizeDigest(sql)
```

Comment:
Seems that we parse twice times due to adding this line. Can we avoid it?

Reply:
Currently is hard to avoid this, but this function was only used for internal SQL, so It won't hurt the user query performance.